### PR TITLE
chore: Temporary - adding SDK preview version in client

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

As part of CLOUDP-257162, this is the first PR made to `CLOUDP-212932-dev` branch so that we have the SDK dev preview available.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,6 +25,7 @@ require (
 	github.com/zclconf/go-cty v1.14.4
 	go.mongodb.org/atlas v0.36.0
 	go.mongodb.org/atlas-sdk/v20231115014 v20231115014.0.0
+	go.mongodb.org/atlas-sdk/v20240530001 v20240530001.0.1-0.20240619083352-d423f3e4cbe1
```
Comment (maastha) [inline go.mod:28]: nit: do we plan to have a task to undo this and upgrade?

**Review by maastha — APPROVED** [review]

**Review by marcosuma — APPROVED** [review]
LGTM - what if this stayed and we found a way to auto-updated it? (Not for now)

File: go.mod (hunk 1 above)
Comment (AgustinBettati) [reply go.mod:28]: Are you referring to eventually removing the preview and adopting the new stable sdk release when ISS is released? Can create a ticket for this.

**Comment by AgustinBettati** [comment]
@marcosuma not sure if I fully understand, are you thinking in terms of having the preview SDK available (and updated) in master branch?

**Comment by lantoli** [comment]
as discussed offline with @AgustinBettati I would rather have the preview as the regular sdk, and have an old sdk. This is because eventually the preview will be the current version so no need to do extra changes

**Pull request merged**